---
title: "homework_02"
author: "Esteban Jorquera"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  html_document: default
  css: readable.min.css
---

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(echo = TRUE)
```

### Practical: Alignment

```{code}

# copies all files from its parent directory to a local work directory
cp /mnt/Timina/bioinfoII/data/alignment/* /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2

# opens a qlogin session
qlogin

# checks module availability 
module avail

# loads a version (0.7.15) of the required program bwa, available from the module list
module load bwa/0.7.15

# indexes the Escherichia_coli_K12_MG1655.fasta file using bwa
bwa index /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/Escherichia_coli_K12_MG1655.fasta

# loads a version (0.11.3) of the required program fastqc, available from the module list
module load fastqc/0.11.3

# manual of the fastqc module 
fastqc -h
#generation of the fastq format from the transcriptional activator FNR from E. coli
fastqc /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/SRX189773_FNR_ChIP.fastq -o output

#the last stpe generates html and zip archives, which we saved on our computers to visualize them
scp ejorquera@dna.lavis.unam.mx:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/output/* /home/esteban/Tarea_2_Results
```

```{code}
#alignment of the Escherichia coli genome reference with the Chip-seq identifier SRX189773
bwa aln /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/Escherichia_coli_K12_MG1655.fasta /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/SRX189773_FNR_ChIP.fastq > /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/output/EColi_FNR_ChIP.sai

#convertion of the sam file of the alignment, so we could open samtools stats
bwa samse /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/Escherichia_coli_K12_MG1655.fasta /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/output/EColi_FNR_ChIP.sai /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/SRX189773_FNR_ChIP.fastq > /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/output/EColi_FNR_ChIP.sam
```

```{code}
module load samtools/1.9

samtools stats /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/output/EColi_FNR_ChIP.sam 
```


## MMusculus


**generates html and zip archives, which we saved on our computers to visualize them
fastqc /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/SRX189773_FNR_ChIP.fastq -o output 
scp ejorquera@dna.lavis.unam.mx:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_2/output/* /home/esteban/Tarea_2_Results **


There are already the refrence genome and index files, so we just need their respective directories:

mm10 reference genome: /mnt/Archives/genome/mouse/mm10/UCSC/bwa-0.7.15-index/index/mm10.fa

mm10 index files: /mnt/Archives/genome/mouse/mm10/UCSC/bwa-0.7.15-index/index/mm10.fa.amb\
/mnt/Archives/genome/mouse/mm10/UCSC/bwa-0.7.15-index/index/mm10.fa.ann\
/mnt/Archives/genome/mouse/mm10/UCSC/bwa-0.7.15-index/index/mm10.fa.bwt\
/mnt/Archives/genome/mouse/mm10/UCSC/bwa-0.7.15-index/index/mm10.fa.pac\
/mnt/Archives/genome/mouse/mm10/UCSC/bwa-0.7.15-index/index/mm10.fa.sa


Generation of a sge script because of the huge amount of sequences our RAM memory wasn't enough

```{code MMusculus.sge}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
#You can edit the scriptsince this line
#
# Your job name
#$ -N Marlon_job

# Send an email after the job has finished
#$ -m e
#$ -M aldarchez26@gmail.com
#
# If modules are needed, source modules environment (Do not delete the next line):
. /etc/profile.d/modules.sh
#
# Add any modules you might require
(module load bwa/0.7.15 ; bwa mem -M -t 8 /mnt/Archives/genome/mouse/mm10/UCSC/bwa-0.7.15-index/index/mm10.fa /mnt/Timina/bioinfoII/data/alignment/mus_musculus_CEBPA_liver_ERR005132.fastq.gz > /mnt/Timina/bioinfoII/marciniega/tarea_02/MMusculus_FNR_ChIP.sam)
```

```{code}
#run of the sge archive to generate our file of interest (sam)
qsub MMusculus.sge

#generation of the samtools stats archive
samtools stats MMusculus_FNR_ChIP.sam > MMusculus_FNR_ChIP.stats

#we open the stats file as follows
nano  MMusculus_FNR_ChIP.stats
```

![image insertion test](%5Chomework_02%5CCEBPA%5Cduplication_levels.png)

##possible justification of the E. coli alignment results:
the highest peak of the Duplicate Sequences graph suggests that there are over-sequence parts of the library, which seems to be justify with the information provided by the overrepresented sequences table where two truSeq Adapters represents almost the 30% of these overrepresentations

